pkg/hook/accessrequesthook/accessrequesthook.go

```diff
@@ -166,6 +166,24 @@ func (h Hook) NoEntitlementAccess(ctx context.Context, cfg *config.Context, inpu
 			req.Justification.Reason = &customReason
 		}
 	}
+
+	for i := range req.Entitlements {
+
+		if input.Duration != nil {
+			req.Entitlements[i].Duration = input.Duration
+			continue
+		}
+
+		if result.DurationConfiguration != nil {
+			if result.DurationConfiguration.DefaultDuration != nil {
+				req.Entitlements[i].Duration = result.DurationConfiguration.DefaultDuration
+			} else {
+				req.Entitlements[i].Duration = result.DurationConfiguration.MaxDuration
+			}
+		}
+
+	}
```

Comment:
```suggestion
	for i := range req.Entitlements {

		if input.Duration != nil {
			req.Entitlements[i].Duration = input.Duration
			continue
		}

		if result.DurationConfiguration != nil {
			if result.DurationConfiguration.DefaultDuration != nil {
				req.Entitlements[i].Duration = result.DurationConfiguration.DefaultDuration
			} else {
				req.Entitlements[i].Duration = result.DurationConfiguration.MaxDuration
			}
		}

	}
```

Only set the duration on the grant when it has been provided as a flag, else, the server will work out the duration.
```suggestion
        if input.Duration != nil {
	        for i := range req.Entitlements {
			req.Entitlements[i].Duration = input.Duration
		}
	}
```